traitsui/tests/test_actions.py

```diff
@@ -204,6 +241,93 @@ def _wx_trigger_button_action(ui):
 
+class TestActionSchemas(BaseTestMixin, unittest.TestCase):
+    def setUp(self):
+        BaseTestMixin.setUp(self)
+
+    def tearDown(self):
+        BaseTestMixin.tearDown(self)
+
+    def _test_actions(self, trigger_action_func):
+        """Template test for wx, qt4, menu, and toolbar testing."""
+        # Behavior: when clicking on a menu or toolbar action,
+        # the corresponding function should be executed
+
+        # create dialog with toolbar adn menu
+        dialog = DialogWithSchema()
+        with reraise_exceptions(), create_ui(dialog) as ui:
+
+            # press toolbar or menu button
+            trigger_action_func(ui)
+
+            # verify that the action was triggered
+            self.assertTrue(dialog.action_successful)
+
+    # ----- Qt tests
+
+    @requires_toolkit([ToolkitName.qt])
+    def test_qt_toolbar_action(self):
+        # Behavior: when clicking on a toolbar action, the corresponding
+        # function should be executed
+
+        # Bug: in the Qt4 backend, a
+        # TypeError: perform() takes exactly 2 arguments (1 given) was raised
+        # instead
+
+        qt_trigger_toolbar_action = partial(
+            _qt_trigger_action, pyface.ui.qt4.action.tool_bar_manager._ToolBar
+        )
+
+        self._test_actions(qt_trigger_toolbar_action)
+
+    @requires_toolkit([ToolkitName.qt])
+    def test_qt_menu_action(self):
+        # Behavior: when clicking on a menu action, the corresponding function
+        # should be executed
+
+        # Bug: in the Qt4 backend, a
+        # TypeError: perform() takes exactly 2 arguments (1 given) was raised
+        # instead
+
+        qt_trigger_menu_action = partial(
+            _qt_trigger_action, pyface.ui.qt4.action.menu_manager._Menu
+        )
+
+        self._test_actions(qt_trigger_menu_action)
+
+    # ----- wx tests
+
+    @unittest.skipIf(
+        not is_mac_os,
+        "Problem with triggering toolbar actions on Linux and Windows. Issue #428.",  # noqa: E501
+    )
+    @requires_toolkit([ToolkitName.wx])
+    def test_wx_toolbar_action(self):
+        # Behavior: when clicking on a toolbar action, the corresponding
+        # function should be executed
+
+        import wx
+
+        def _wx_trigger_toolbar_action(ui):
+            # long road to get at the Id of the toolbar button
+            toolbar_control = ui.control.GetToolBar()
+            toolbar_item = toolbar_control.tool_bar_manager.groups[0].items[0]
+            toolbar_item_wrapper = toolbar_item._wrappers[0]
+            control_id = toolbar_item_wrapper.control_id
+
+            # build event that clicks the button
+            click_event = wx.CommandEvent(
+                wx.wxEVT_COMMAND_TOOL_CLICKED, control_id
+            )
+
+            # send the event to the toolbar
+            toolbar_control.ProcessEvent(click_event)
+
+        self._test_actions(_wx_trigger_toolbar_action)
+
+    # TODO: I couldn't find a way to press menu items programmatically for wx
```

Comment:
I tested manually and it worked as expected

Reply:
I had to add `DialogWithSchema().configure_traits()` in the `if __name__ == "__main__"` below.  Maybe it is worth adding this for more explicit manual testing since there isn't a unittest?